### PR TITLE
Fix dependency issues in CircleCI build

### PR DESCRIPTION
**Pull request by vasudevanv**
Added a requirements.txt file which will be used by CircleCI in
setting up the virtual environment. Hopefully this will solve the
build failures due to dependencies.

### CODE REVIEW

**Comment by dhimmel** [comment]
One downside is that we now will have requirements and version info in two places.

I wonder if there is another solution to fix [the bug](https://github.com/cognoma/cognoml/pull/19#issuecomment-264520338). Would any of the following `pre` steps work

+ Just `pip install numpy==1.11.1`, so we're minimally repetative
+ `pip install .` -- use setuptools rather than distutils to install `cognoml`

Or keep the current pre step but have `setup.py` read `requirements.txt` to populate `install_requires`.

**Comment by vasudevanv** [comment]
I realized this the moment I hit the submit on the pull request. Indeed a downside. Anyways the build is still failing. I think we can give the just `pip install numpy==1.11.1` a try.

**Comment by vasudevanv** [comment]
@dhimmel's suggestion of pip installing the bare essentials works. I removed the requirements.txt file. I needed to do add some extra commands to the pre and machine-post sections to make sure scipy builds within time.

**Comment by dhimmel** [comment]
Nice work getting the CircleCI running. 

It [looks like](https://circleci.com/gh/cognoma/cognoml/11) CircleCI is running nosetests rather than pytest. @vasudevanv do you want to deal with that in a separate PR or do it here?

**Comment by vasudevanv** [comment]
Haha, I was going to ask you that. I will fix it in this PR.

File: circle.yml [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+  pre:
+    - pip install --upgrade pip
+    - pip install Cython --install-option="--no-cython-compile"
+    - pip install numpy==1.11.1 
+    - pip install scipy==0.17.1
+    - pip install pytest==3.0.4
+
+test:
+  override:
+    - py.test -v
```
Comment (dhimmel) [inline circle.yml]: What do you think about `--verbose` rather than `-v`? I like the long names for readability.

**Review by dhimmel — APPROVED** [review]
Want to

File: circle.yml (hunk 1 above)
Comment (vasudevanv) [reply circle.yml]: I personally do not have a preference. Will update it.

**Pull request merged**